src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -229,7 +225,16 @@ func runClientProxy(cfg *config.Config) {
 
 	go func() {
 		addExitFunc(pacOff)
-		client.ListenAndServe(pacOn)
+		client.ListenAndServe(func() {
+			pacOn()
+			if showui {
+				// Launch a browser window with Lantern but only after the pac
+				// URL and the proxy server are all up and running to avoid
+				// race conditions where we change the proxy setup while the
+				// UI server and proxy server are still coming up.
+				ui.Show()
+			}
+		})
```

Comment:
Yeah, it makes sense to wait to launch the browser window only after we've set Lantern as the system proxy. Nice change! Works well for me on Chrome, FF, and IE.